controllers/tab/settings/archiving/form/ArchivingForm.inc.php

```diff
@@ -39,6 +69,46 @@ function __construct($wizardMode = false) {
 	function getLocaleFieldNames() {
 		return array('lockssLicense', 'clockssLicense');
 	}
+
+	/**
+	 * @see Form::execute()
+	 * @param $request PKPRequest
+	 */
+	function execute($request) {
+		parent::execute($request);
+
+		$versionDao = DAORegistry::getDAO('VersionDAO');
+		$product = $versionDao->getCurrentVersion("plugins.generic", "pln", true);
+		$categoryDir = PLUGINS_PREFIX . 'generic';
+		if (isset($product)) {
+			$file = $product->getProduct();
+
+			$plugin =& PluginRegistry::_instantiatePlugin('generic', $categoryDir, $file, $product->getProductClassname());
```

Comment:
`PluginRegistry::_instantiatePlugin` is a private function (we used to use a `_` prefix to indicate this, while we were supporting versions of PHP that didn't support declaring it explicitly). It shouldn't be called outside the `PluginRegistry` class.

Reply:
Also, references are no longer necessary on objects in general; don't use them unless they're required.

Reply:
Fixed.